services/provider/server/server.go

```diff
@@ -584,6 +581,25 @@ func decodeAndValidateTicket(ticket string, pubKey *rsa.PublicKey) (*services.On
 	return &ticketData, nil
 }
 
+func decodeTicket(ticket string) ([]byte, []byte, error) {
+	ticketArr := strings.Split(string(ticket), ".")
+	if len(ticketArr) != 2 {
+		return nil, nil, fmt.Errorf("invalid ticket")
+	}
+
+	message, err := base64.StdEncoding.DecodeString(ticketArr[0])
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode onboarding ticket message: %v", err)
+	}
+
+	signature, err := base64.StdEncoding.DecodeString(ticketArr[1])
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode onboarding ticket signature: %v", err)
+	}
```

Comment:
Why decode the signature always, only a single call site actually cares about the signature.
It seems to me the original `decodeAndValidate` was the current approach and should be used when you need to decode and validate, and for the second call site you just want a `readStorageClusterUid` method that does not need to read the signature or validate